server/__test__/course.test.js

```diff
@@ -0,0 +1,49 @@
+const request = require('supertest');
+const connection = require('../src/database/config/connection');
+const dbBuild = require('../src/database/config/build');
+
+const app = require('../src/app');
+
+describe('get all courses', () => {
+  beforeAll(() => dbBuild());
+  afterAll(() => connection.end());
+
+  it('get all courses by category id "1"', async () => {
+    expect.assertions(5);
+    await request(app)
+      .get('/api/v1/1/courses')
+      .set('Accept', 'application/json')
+      .expect(200)
+      .then((response) => {
+        const data = response.body;
+        expect(data[0].id).toBe(1);
+        expect(data[0].title).toBe('computer');
+        expect(data[0].image).toBe(
+          'https://uniweb.qwebbuilder.com.ng/images/onlinecourses.jpg'
+        );
+        expect(data[0].rate).toBe('3.4');
+        expect(data[0].source).toBe('udemy');
```

Comment:
**you don't need to use promise with `async await` , fix it as the following :**
```suggestion
  const response =   await request(app)
      .get('/api/v1/1/courses')
      .set('Accept', 'application/json')
      .expect(200);
        const data = response.body;
        expect(data[0].id).toBe(1);
        expect(data[0].title).toBe('computer');
        expect(data[0].image).toBe(
          'https://uniweb.qwebbuilder.com.ng/images/onlinecourses.jpg'
        );
        expect(data[0].rate).toBe('3.4');
        expect(data[0].source).toBe('udemy');
```